master/buildbot/steps/source/git.py

```diff
@@ -583,9 +583,13 @@ def checkBranchSupport(self):
         stdout = yield self._dovccmd(['--version'], collectStdout=True)
 
         gitInstalled = False
+        version = "0.0.0"
         if 'git' in stdout:
             gitInstalled = True
-        version = stdout.strip().split(' ')[2]
+            try:
+                version = stdout.strip().split(' ')[2]
+            except IndexError:
+                gitInstalled = False
```

Comment:
I think we should also handle the case where split(' ')[2] exists but do not return a correct version

Reply:
I don't know what an incorrect version is.  This code https://github.com/buildbot/buildbot/blob/master/master/buildbot/steps/source/git.py#L589

uses LooseVersion.  According to http://epydoc.sourceforge.net/stdlib/distutils.version.LooseVersion-class.html ,
with LooseVersion, there is no such thing as an invalid version number.

Reply:
would "a1" be considered an invalid version #?

Reply:
`LooseVersion` accepts anything as a version, including "a1"

Reply:
okay. If there is another use case later, we can fix it at that time.